statefun-sdk/src/lib.rs

```diff
@@ -299,3 +300,18 @@ impl Display for EgressIdentifier {
         write!(f, "EgressIdentifier {}/{}", self.namespace, self.name)
     }
 }
+
+/// Errors that can occur during function invocation.
```

Comment:
Hi, may be it would make sense to move `InvocationError` and `HyperTransportError` in a separate module `error.rs`.
Then the sdk users can then import it is `use statfun_rust::error::InvocationError;` and so on.

For reference:
https://github.com/seanmonstar/warp/blob/master/src/error.rs
https://github.com/dani-garcia/bitwarden_rs/blob/master/src/error.rs
https://github.com/actix/actix-web/blob/master/src/error.rs

Reply:
Thanks for the pointers. I think they all have somewhat mixed approaches. Warp has the errors in a separate file but re-exports them in `lib.rs` so to users it looks like they're in the root package. Bitwarden is not a library but also re-exports the error. actix-web does not re-export the error.

So I can move them to a separate file but would maybe re-export them at the root? What do you think?

I would not move `HyperTransportError` and `InvocationError` to the same module because they are errors for different components.

Reply:
> So I can move them to a separate file but would maybe re-export them at the root? What do you think?

I think this would be a very nice approach. This would keep the code cleaner + errors then can be used properly from the perspective of sdk as well.

Reply:
I pushed a commit for that change.